Porting Cardinal Station from Beestation

Pikamaster07 opened the pull request:
## About The Pull Request
I wanted to port Cardinal station from beestation's map rotation, I Have been at work fixing it up for bubber myself the past few weeks.

As of now, it is completely working to my knowledge and will run a full round without issue. all shuttles (the normal ones at least) work properly, all machines have been properly added in and such. it should be completely up to date for bubbers standards.

## Why It's Good For The Game
People like new maps right? I've personally played on this map many times when I was on Beestation and its considered one of the best maps in their rotation, especially during highpop. 

While I personally am NOT someone who understands game balance and such, I did add in certain personal changes to the map out of the though of them being at least fun to interect with and possibly creating interesting situations. I tried to not make anything gamebreaking or such put in but I do hope ive at least brought something that people will look forward to playing on rather than dread.

## Proof Of Testing
<details>
<summary>Screenshots/Videos</summary>

<img width="1048" height="976" alt="image" src="https://github.com/user-attachments/assets/e26baf9b-42c8-4435-9172-bf780834f810" />
<img width="635" height="632" alt="image" src="https://github.com/user-attachments/assets/16df22cf-8cc6-4fe3-93b4-cc33a8f1620e" />
<img width="728" height="744" alt="image" src="https://github.com/user-attachments/assets/ce51f72c-6943-49c5-bc15-9607a4949118" />
<img width="1355" height="403" alt="image" src="https://github.com/user-attachments/assets/370cc274-cdc4-4dbc-95f6-53b04f7cd153" />
(if i need more images or better ones do tell)


</details>

## Changelog
:cl:
add: CardinalStation

/:cl:

Pikamaster07 committed:
this is the map

TheEmber committed:
Merge pull request #1 from Pikamaster07/master

this is the map

TheEmber committed:
added map to other files

TheEmber committed:
forgo json

TheEmber committed:
shuttles

TheEmber committed:
name

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 committed:
Add files via upload

Pikamaster07 commented:
Just added in a mecha hanger in security. Also added in more random items in maints, proper food vendors around the map, fire and oxygen lockers that were missing, removed one of the bitrunning dens (oops, didnt know 3 was the limit), and maybe a few more minor things I cant remember. 

Things still missing: Door access (ill try working on this next), disposal pipes are still fucky (the.. filter pipes? i dont know the name, anyways some pipes are missing still from the port and I dont know how to fix them up to work. hoping someone else can help do that.), and roundstart shuttles (which I also have no clue how to get working, I have the shuttles the map used as well, but they need some fixing from the port over. Again, really hoping someone else can work on this or help. Oh right uhh arrival terminal needs work too.)

I could be forgetting about something but I think im close to being ready. Finishing these last three things and I cant think of anything else the map is missing.

TheEmber committed:
Merge branch 'master' of github.com:Pikamaster07/Bubberstation-CardinalPort into cardinal

TheEmber committed:
shuttles

TheEmber committed:
evac shuttle thru strongdmm

TheEmber committed:
fixed areas

TheEmber committed:
shuttle floors

TheEmber committed:
will shuttle work?

TheEmber committed:
cargo shuttle again

TheEmber committed:
another attempt to get cargo shuttle working

Pikamaster07 committed:
...I forgot what i added.

i think i put it in the pr draft...  its very late.

Pikamaster07 committed:
map updated

updated areas to match latest bubber area pathing. finished door access for science.

TheEmber committed:
Revert "another attempt to get cargo shuttle working"

This reverts commit b95a09eb1f5ed297defd54710f516a95f6544393.

TheEmber committed:
Revert "cargo shuttle again"

This reverts commit af0d4fd07c9698f7f6f75b14709bc2a630034b46.

TheEmber committed:
Revert "will shuttle work?"

This reverts commit 6924828181bc2d43ddf22eae8bafb9a0d424f05f.

TheEmber committed:
Merge branch 'master' of github.com:Pikamaster07/Bubberstation-CardinalPort into cardinal

Pikamaster07 committed:
map updated

cant remember changes. more door accesses added, maints stuff... uhmmm, fixed other minor things. there were invisible cables... very annoying but shouldnt affect power anymore

TheEmber committed:
Merge branch 'Bubberstation:master' into cardinal

TheEmber committed:
delete to resolve conflict?

TheEmber committed:
Merge branch 'master' of https://github.com/Pikamaster07/Bubberstation-CardinalPort into cardinal

Pikamaster07 committed:
Merge pull request #1 from TheEmber/cardinal

Cardinal rulessss

Pikamaster07 committed:
Door access and misc

Pikamaster07 committed:
map update

Pikamaster07 committed:
map update

Pikamaster07 committed:
removing extra secmed locker automapper

Pikamaster07 committed:
map update

Pikamaster07 committed:
map update

Pikamaster07 committed:
map update

messing around with supermatter stuff

Pikamaster07 committed:
map update

Pikamaster07 committed:
map update

Pikamaster07 committed:
wall healers

Pikamaster07 committed:
map update

going to work on windoors next

Pikamaster07 committed:
all important windoors added

maybe im still missing a few from the original map but all the windoors an front desks are done. in fact... I have very little left to do

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 committed:
map update

Pikamaster07 committed:
map update

I missed a few windoors, fixed now

Pikamaster07 committed:
map update

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 committed:
map update?

Pikamaster07 committed:
im going to try adding in more shuttles

maybe you can do something with this ember?

shayoki committed:
Mail Disposals

mail chutes and NTR office disposal fix

shayoki committed:
Merge branch 'master' of https://github.com/Pikamaster07/Bubberstation-CardinalPort

Pikamaster07 committed:
map update

Pikamaster07 committed:
map update

TheEmber committed:
cargo shuttle

TheEmber committed:
evac shuttle

TheEmber committed:
mining

TheEmber committed:
labor shuttle

Pikamaster07 committed:
Merge pull request #2 from TheEmber/cardinal

Cardinal

Pikamaster07 committed:
finished interiors of emergency and cargo shuttles

Pikamaster07 committed:
public mining shuttle?

Pikamaster07 committed:
thruster fix?

Pikamaster07 committed:
broke stuff, hold on

Pikamaster07 committed:
thruster fix pls

Pikamaster07 committed:
OOOOPS

Pikamaster07 committed:
fix please :(

Pikamaster07 committed:
there is no way this actually works

Pikamaster07 committed:
shuttle changes

Pikamaster07 committed:
ready for review

yeah I couldnt get the exploration shuttle working, someone else will probably get it working and props to whoever it is.

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 committed:
map error fix attempt

Pikamaster07 committed:
minor fixes, last minute cosmetic changes.

Pikamaster07 committed:
4 lights, im serious

shayoki committed:
CI

makes it votable on locals and fixes some CI errors

shayoki committed:
removes all req_access_txt

no more oldcode

shayoki committed:
remove req_one_access_txt

shayoki committed:
Update CardinalStation.dmm

id -> id_tag

shayoki committed:
Update CardinalStation.dmm

shayoki committed:
remove announcementconsole var

more oldcode replaced with mapping helpers

shayoki committed:
CI fixes: more edition

pathing scrips are kinda awesome

shayoki committed:
some shuttle fix and area fix

shayoki committed:
more shuttle fix

shayoki reviewed (changes requested):
Hi there. Please resolve some the linter errors that were picked up by the unit tests. [Notably `/datum/unit_test/firedoor_regions` and `/datum/unit_test/cable_powernets`](https://github.com/Bubberstation/Bubberstation/actions/runs/19783152879?pr=4932) . These display the coordinates on the failed lines on the error messages on what to look for when working on the map. Good luck!

Pikamaster07 committed:
shuttle map error fix 1

Pikamaster07 committed:
shuttle map errors 2

Pikamaster07 committed:
Map errors 1

Pikamaster07 committed:
map errors 2

Pikamaster07 committed:
Map errors 3

Pikamaster07 committed:
Map errors 4

Pikamaster07 committed:
Map errors 4

Pikamaster07 committed:
Map errors 5 (this is going to take forever :pensive:)

Pikamaster07 committed:
map errors 7

RAHHHHHHH

Pikamaster07 committed:
WORK YOU STUPID FUCK

Pikamaster07 committed:
no you SHIT

Pikamaster07 committed:
map errors 8

Pikamaster07 committed:
map errors 9

Pikamaster07 committed:
map errors 10

Pikamaster07 committed:
fuck windoors

Pikamaster07 committed:
windoors suck ass

Pikamaster07 commented:
<img width="1272" height="630" alt="image" src="https://github.com/user-attachments/assets/eeaa998a-8540-4905-9f20-16b77fa12113" />

hey, i dont know how to fix this error. all mapping errors are fixed but i dont know what THIS is or how to fix it

Pikamaster07 committed:
Update CardinalStation.dmm (PLEASE WORK)

Pikamaster07 committed:
fuck you, COMPILE

Pikamaster07 committed:
ughhhhh

Pikamaster07 committed:
Update CardinalStation.dmm

Pikamaster07 committed:
Update CardinalStation.dmm

Pikamaster07 committed:
Update CardinalStation.dmm

Pikamaster07 committed:
Update CardinalStation.dmm

Pikamaster07 committed:
Update CardinalStation.dmm

Pikamaster07 committed:
so close (I HOPE)

Pikamaster07 committed:
integration error final

Pikamaster07 committed:
final integration

Pikamaster07 committed:
HOW IS THERE MORE!?? 

FUUUUUCK YOU

Pikamaster07 committed:
Update CardinalStation.dmm. again. fuck mapping

Pikamaster07 committed:
work

Pikamaster07 committed:
Update CardinalStation.dmm WORK DAMN IT

Pikamaster07 committed:
Delete _maps/map_files/CardinalStation/CardinalStation(2).dmm

Pikamaster07 committed:
DAMN YOU

Pikamaster07 committed:
dont fuck with me asshole

Pikamaster07 committed:
Happy new years!

Pikamaster07 commented:
integration tests still failing for some reason, cant fix it because I cant even figure out why. everything else works AFAIK and have seen. submitting this again. hopefully this is good enough now. happy new years

Pikamaster07 committed:
integration fix

Pikamaster07 committed:
integration fix

Pikamaster07 committed:
Delete _maps/map_files/CardinalStation.dmm

Extra, annoying extra. My bad

Pikamaster07 committed:
oops cargo error

Pikamaster07 committed:
requested changes

Pikamaster07 committed:
oops. fixed now

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
Add files via upload

SapphoQueer commented:
bridge has no fireaxe
bitrunning shit can't be deposited?

R0bro commented:
Airlocks in the aux base construction site aren't accessible to science personnel.

Nairbhaon commented:
Telecomms has the wrong airtight flaps installed

xPokee commented:
The recycler is facing the wrong direction.

Nairbhaon commented:
The departures sec should be set to just sec. Right now it is set to maints and service, so lots of people that aren't sec can get in.

BurgerLUA commented:
General opinion issues:
- Donkpocket vendor shouldn't be a station spawn, as it contains gondola pockets.
- There is a pool of lava in maint, behind a door. This is dangerous for people who have lag.

General Fact issues:
- Exosuit lathe in maint has a silo link. It should be the maintenance subtype.

Access issues: 
- Security checkpoint maint door in evac has incorrect access.
- Science desk windoors has incorrect access.

R0bro commented:
The cytology lab has no air alarm.
There are bio bags and virology lockers in xenobiology which is incorrect
The science front desk windoors are all access
That is all for science that I haven't already mentioned

BurgerLUA commented:
Forgot to post this 
<img width="726" height="409" alt="image" src="https://github.com/user-attachments/assets/ce4c0dd3-4974-4974-ba64-f61ce0128410" />

Assistant access plasma canister. I think Tram has one so it might be fine.

Pikamaster07 committed:
fixed what i could.

Pikamaster07 committed:
oops

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
INTEG FIX

Pikamaster07 committed:
Delete _maps/CardinalStation.dmm

Pikamaster07 commented:
> Minor nitpick but when on the lobby screen it shows up as Card Station instead of Cardinal Station, stations should probably be called by their actual names.

i know yeah,  quirk of the original map. ill fix it later

ibexgoetia commented:
My two cents as a medical and research main;

I think that the medbay's actual operating area is too small. There are two stasis beds already built, but there isn't really a space to build new beds without changing the architecture entirely. I feel like some walls might have to be removed and maybe part of the waiting area be turned into surgery rooms, because seeing as how this is a station meant for higher playercounts, it's going to get very crowded very easily.

There is also genetics attached to medical, as a holdover from older code.
<img width="400" alt="image" src="https://github.com/user-attachments/assets/e5334e71-2da5-4e7c-9618-fddf00cbe365" />

Robotics should have its desk closer to that of R&D, following how most other maps do it. Makes it a lot easier for roboticists to multitask doing basic research stuff like research, printing items, etc.

Pikamaster07 commented:
> My two cents as a medical and research main;
> 
> I think that the medbay's actual operating area is too small. There are two stasis beds already built, but there isn't really a space to build new beds without changing the architecture entirely. I feel like some walls might have to be removed and maybe part of the waiting area be turned into surgery rooms, because seeing as how this is a station meant for higher playercounts, it's going to get very crowded very easily.
> 
> There is also genetics attached to medical, as a holdover from older code. <img alt="image" width="400" src="https://private-user-images.githubusercontent.com/146444134/531621986-e5334e71-2da5-4e7c-9618-fddf00cbe365.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc0NzM4ODYsIm5iZiI6MTc2NzQ3MzU4NiwicGF0aCI6Ii8xNDY0NDQxMzQvNTMxNjIxOTg2LWU1MzM0ZTcxLTJkYTUtNGU3Yy05NjE4LWZkZGYwMGNiZTM2NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEwM1QyMDUzMDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNzU2N2FkZDIwMmNhYjYxODE2ZWQ5MDAxMjhlYmMzZTA1OTM3MDg2MzFjYjhiNDc4MWFiMWYzYWY5ZTI4OTQ3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.B4OhsDieXOlvN8U0IqByInkyOunS1cE9InYRSZp8l3M">
> 
> Robotics should have its desk closer to that of R&D, following how most other maps do it. Makes it a lot easier for roboticists to multitask doing basic research stuff like research, printing items, etc.

Regarding medical: looking back at it now, yeah I can see that. Its rather easy to get an extra stasis bed put in each of the little ‘corners’ by the existing ones, but even then I believe that is insufficient for Cardinals size. Resizing it is not exactly easy to do, but ill see if i could at least double what i can. Size is also why I gave it 3 ORs. 

Regarding genetics: Yeah its old, but im keeping that. My thought is that it would increase genetics traffic since not too many use genetics. The other reason is that I REALLY don’t want to scramble science around to make room for it. The final reason is that I like where it is.

regarding robotics: not sure what this means exactly.

BingusTheClown commented:
Couple of things from just playing around:

Medbay:
- Virology is massive, and can probably be downsized, it doesn't need a break room and 2-3 sets of machines, and the patient rooms inside are on the farrr end, probably could be swapped to have layout rotated. (cutting off need for a break room) 
- Surgery has 3 organ holders, can probably be made 2, between the actual walls instead?
- Old Cloning has no purpose (others have probably said that, so ignore me for now lol) 
- The lab to do all the chem plumbing is really far out of the way, and has no way for people to know what is going on
- working Cloner in freezer
- unsure what bubber policy around sleepers in medical is
- Do we need a second med station above the bar? (considering the bar is quite small)
- Some of the hallways to the freezer are mislabeled (one says surgery overview, or something of that nature)

Pikamaster07 committed:
fixed a few things. reworked exam room

Pikamaster07 commented:
> > > Robotics should have its desk closer to that of R&D, following how most other maps do it. Makes it a lot easier for roboticists to multitask doing basic research stuff like research, printing items, etc.
> > 
> > 
> > regarding robotics: not sure what this means exactly.
> 
> On most stations, Robotics, R&D front desk, and genetics are pretty close -- it's generally a trend that you can see either the genetics or robotics dept. if you can see the front desk, if not all three. Here's how Science looks on other stations: <img alt="image" width="1563" height="990" src="https://private-user-images.githubusercontent.com/192442917/531675874-dcb375cc-5f27-4e15-a403-8f45d9693a52.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc0OTUxMTYsIm5iZiI6MTc2NzQ5NDgxNiwicGF0aCI6Ii8xOTI0NDI5MTcvNTMxNjc1ODc0LWRjYjM3NWNjLTVmMjctNGUxNS1hNDAzLThmNDVkOTY5M2E1Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEwNFQwMjQ2NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MDY3ODU2ODFiMDU1MjkwNzI1ZjZhZGE3MzFmOGMzM2FhODZmZDgzZDIzZTI5MzczZDZlNjcwMzRiZGEwYjY0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ks4usHtIZDJPnaFsX0YZnpqcCvClkrnSBTz94lvdW0g">
> 
> This is usually a good idea to default to since, if someone's at the science front desk but doesn't see any scientists, they won't have to go far to bug robotics or genetics in their depts to ask them for a general scientist thing (and relatedly, robotics and genetics won't need to travel far to get their equipment from the R&D front desk).
> 
> I won't say no to genetics being in medbay for now, I'd like to see if the experiment pans out and people use genetics more, but I do feel like it would be smarter to place that closer to the rest of R&D long-term.

Ahhh i understand now. thanks for letting me know

Pikamaster07 committed:
cam console change

Pikamaster07 committed:
fixed a reused area

Pikamaster07 committed:
oops

Pikamaster07 committed:
damn atmos error

xPokee committed:
Update _basemap.dm

xPokee committed:
Revert "Update _basemap.dm"

This reverts commit 0f593922545fe6c7e85b15e9e4d20219bffb5b6b.

sippykot commented:
can we unstale this

Pikamaster07 committed:
Update CardinalStation.dmm, autoname test

Pikamaster07 committed:
Update CardinalStation.dmm. improved camera autoname

Pikamaster07 committed:
reverted advanced camera console change (what was i thinking)

Pikamaster07 committed:
Merge branch 'master' into master

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 committed:
changed borg spawns around

made all borgs spawn on the sat because I like spawning next to all the borg friends and it looks cute :>

shayoki committed:
Merge branch 'master' into master

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 committed:
foliage, my bad!

Pikamaster07 committed:
Security now has cells (plus that one thing in armory yes i did add that yes i also think its super cool)

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 committed:
oop, bad offsets on a cryox beaker in med. should be good now

Pikamaster07 committed:
ughggghg pipenets...

TealSeer commented:
- The cluster of four doors leading into the engineering foyer are all just named Airlock
- Ditto on the entrance to science
- Ditto for primary tool storage
- Ditto for Theatre
- Ditto for Courtroom

Pikamaster07 commented:
yes there are a lot of doors I need to rename, ill get to it all eventually. thanks for the heads up though

MrGloopy commented:
grinder in botany is inside the window, as I said ingame. Needs to be moved one tile south. Chairs appear to be missing from the diner area.

Pikamaster07 committed:
tore the doors off for now, im not dealing with this

Pikamaster07 committed:
lot of map fixes

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 committed:
im, very tired

Pikamaster07 committed:
ffffuck youuuu

Pikamaster07 committed:
try again

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 committed:
piece a SHIET

Pikamaster07 committed:
ripped the doors off, AGAIN. work you fuck

Pikamaster07 committed:
i cant believe i missed the bitrunning order machine

Pikamaster07 committed:
windoors

Pikamaster07 committed:
fixed... fuck a lot o f things i tink

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 commented:
genuinely no idea how to fix the belts thing. I REALLY need someone to find out whats causing it because its some fucky shit that I cannot figure out for the life of me

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
power issue in AIsat resolved

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 committed:
Update CardinalStation.dmm (try again)

Pikamaster07 committed:
Update CardinalStation.dmm (WORK)

Pikamaster07 committed:
missed a lever

Pikamaster07 committed:
good changes

Pikamaster07 committed:
Merge branch 'Bubberstation:master' into master

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
maint alters

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
experimental cloner machines removed

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
FUCK I SHOULD HAVE COMMITED SOONER

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
Add files via upload

Pikamaster07 committed:
Add files via upload

SapphoQueer commented:
Un-testmerging this until a lot of the more extravagant loot is removed from the map and the maptainer looks over it again.

Pikamaster07 committed:
new space ruin area

Pikamaster07 committed:
Add files via upload

Pikamaster07 commented:
> Un-testmerging this until a lot of the more extravagant loot is removed from the map and the maptainer looks over it again.

Do I stop working on this until thats done? The lookover.

sippykot commented:
genuinely just slowly removing soul from the map at this point
give us back the cloning setup...
